x/swaprouter/keeper.go

```diff
@@ -0,0 +1,103 @@
+package swaprouter
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	gogotypes "github.com/gogo/protobuf/types"
+
+	"github.com/osmosis-labs/osmosis/v13/osmoutils"
+	"github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+type Keeper struct {
+	storeKey sdk.StoreKey
+
+	gammKeeper           types.SwapI
+	concentratedKeeper   types.SwapI
+	poolIncentivesKeeper types.PoolIncentivesKeeperI
+	bankKeeper           types.BankI
+	accountKeeper        types.AccountI
+	communityPoolKeeper  types.CommunityPoolI
+
+	poolCreationListeners types.PoolCreationListeners
+
+	routes map[types.PoolType]types.SwapI
+
+	paramSpace paramtypes.Subspace
+}
+
+func NewKeeper(storeKey sdk.StoreKey, paramSpace paramtypes.Subspace, gammKeeper types.SwapI, concentratedKeeper types.SwapI, bankKeeper types.BankI, accountKeeper types.AccountI, communityPoolKeeper types.CommunityPoolI) *Keeper {
+	// set KeyTable if it has not already been set
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	routes := map[types.PoolType]types.SwapI{
+		types.Balancer:     gammKeeper,
+		types.StableSwap:   gammKeeper,
+		types.Concentrated: concentratedKeeper,
+	}
+
+	return &Keeper{storeKey: storeKey, paramSpace: paramSpace, gammKeeper: gammKeeper, concentratedKeeper: concentratedKeeper, bankKeeper: bankKeeper, accountKeeper: accountKeeper, communityPoolKeeper: communityPoolKeeper, routes: routes}
+}
```

Comment:
We aren't initializing this keeper anywhere yet right? Just ensuring there are in fact no state breaks

Reply:
Yes, that's correct!